test/fuzz/p2p/secret_connection/read_write.go

```diff
@@ -0,0 +1,114 @@
+package secretconnection
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+
+	"github.com/tendermint/tendermint/crypto/ed25519"
+	"github.com/tendermint/tendermint/libs/async"
+	sc "github.com/tendermint/tendermint/p2p/conn"
+)
+
+func Fuzz(data []byte) int {
+	if len(data) == 0 {
+		return -1
+	}
+
+	fooConn, barConn := makeSecretConnPair()
+	n, err := fooConn.Write(data)
+	if err != nil {
+		panic(err)
+	}
+	dataRead := make([]byte, n)
+	m, err := barConn.Read(dataRead)
+	if err != nil {
+		panic(err)
+	}
+	if !bytes.Equal(data[:n], dataRead[:m]) {
+		panic(fmt.Sprintf("bytes written %X != read %X", data[:n], dataRead[:m]))
+	}
+	return 1
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// COPY FROM p2p/conn/secret_connection_test.go
+///////////////////////////////////////////////////////////////////////////////
```

Comment:
Why do we need this?

Reply:
Because of the handshake. We need an opposite side otherwise the handshake would just stall forever (if other side is nop). After the handshake is completed we can read and write bytes.

Reply:
We need the comment for this?

Reply:
Oh, you meant the comment. We don't need it per se, but I figured it would be nice to point to "original" location, so that if something changes, the developer could just copy the code from secret_connection_test.go. Does that make sense?